py-polars/polars/_plr.py

```diff
@@ -50,10 +57,23 @@ def pl32() -> None:
                 msg = f"Invalid value for `POLARS_PREFER_PKG` variable: '{_prefer}'"
                 raise ValueError(msg) from None
 
-        for i, pkg in enumerate(preference):
+        for pkg in preference:
             try:
                 pkg()
+
+                if sys.modules[__name__].__version__ != pkg_version:
+                    import warnings
+
+                    warnings.warn(
+                        f"Polars' Rust module version '{sys.modules[__name__].__version__}' did not match version of Python package '{pkg_version}'",
```

Comment:
I think this message should mention that as a result of this, this module was skipped.